src/schemas/json/typst.json

```diff
@@ -0,0 +1,245 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$id": "https://json.schemastore.org/typst.json",
+  "title": "Typst Package Manifest",
+  "description": "Schema for typst.toml package manifest files, used by the Typst package manager and universe.",
+  "type": "object",
+  "required": ["package"],
+  "properties": {
+    "package": {
+      "type": "object",
+      "description": "Contains metadata about the package.",
+      "required": [
+        "name",
+        "version",
+        "entrypoint",
+        "authors",
+        "license",
+        "description"
+      ],
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The package's identifier in its namespace. Should use kebab-case if it contains multiple words. Should not contain 'typst'.",
+          "pattern": "^[a-z0-9][a-z0-9\\-]*$",
+          "examples": ["my-package", "example", "uni-stuttgart-template"]
+        },
+        "version": {
+          "type": "string",
+          "description": "The package's version as a full major-minor-patch triple (following SemVer). See https://semver.org/",
+          "pattern": "^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?$",
```

Comment:
I'm having a bit of a hard time understanding everything that is going on here, but Typst doesn't support pre-release tags (at least for now), so I think having `^(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)$` would be enough.